shared-bin/src/logger.rs

```diff
@@ -116,18 +120,31 @@ pub fn init(
         None
     };
 
+    let (file_log, guard) = if log_to_file {
+        let file_appender =
+            tracing_appender::rolling::never(data_dir, format!("{service_name}.log"));
```

Comment:
Is there a way to say what the max size of this file can be? I'm afraid that we will flood the disk with logs if this is unbound.

Reply:
We could roll the log file daily.

Reply:
Would that delete the old file automatically?

Reply:
unfortunately not.. there is a github project which would allow to create a new log file after a certain size has exceeded, but it does not have a lot of attention https://github.com/Axcient/rolling-file-rs (last commit 2 years ago and just a few stars)

tracing appender is from the tokio team but only supports a limited set of rollover scenarios 
- minutely
- hourly
- daily
- never
https://github.com/tokio-rs/tracing/tree/master/tracing-appender

We could go for daily log files, but I feel like that will produce a lot of small files on the long run, which is also not very nice. Can we estimate how much logs will be produced as to get a feeling if never would be a viable option? respectively can we limit the log level so that the file will not grow unmanageable?

Reply:
We used https://github.com/tokio-rs/tracing/tree/master/tracing-appender in the XMR-BTC project with a `never` appender:

https://github.com/comit-network/xmr-btc-swap/blob/9e85637419594f69d05e0d03767d66b07001a0ae/swap/src/cli/tracing.rs#L19

We should be able to [define a writer for `tracing::appender`](https://github.com/tokio-rs/tracing/tree/master/tracing-appender) 

| Using any type implementing [std::io::Write](https://doc.rust-lang.org/std/io/trait.Write.html) in a non-blocking fashion.

When looking around I found https://github.com/BourgondAries/file-rotate which I think offers what we want and seems to be still under active development. As far as I could see it implements `std::io::Write` we we should be able to use it with `tracing-appender`. Haven't given it a try though.

Reply:
Doesn't have many stars either, but I will give it a try.